Wikipedia/Code/AppearanceSettingsViewController.swift

```diff
@@ -0,0 +1,245 @@
+import UIKit
+
+@objc public protocol WMFAppearanceSettingsViewControllerDelegate {
+    func themeCangedInController(_ controller: AppearanceSettingsViewController, theme: Theme)
+}
+
+protocol AppearanceSettingsItem {
+    var title: String? { get }
+}
+
+struct AppearanceSettingsSwitchItem: AppearanceSettingsItem {
+    let title: String?
+}
+
+struct AppearanceSettingsCheckmarkItem: AppearanceSettingsItem {
+    let title: String?
+    let theme: Theme
+    let checkmarkAction: () -> Void
+}
+
+struct AppearanceSettingsCustomViewItem: AppearanceSettingsItem {
+    let title: String?
+    let viewController: UIViewController
+}
+
+struct AppearanceSettingsSection {
+    let headerTitle: String
+    let footerText: String?
+    let items: [AppearanceSettingsItem]
+}
+
+@objc(WMFAppearanceSettingsViewController)
+open class AppearanceSettingsViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    var sections = [AppearanceSettingsSection]()
+    
+    fileprivate var theme = Theme.standard
+    
+    open weak var delegate: WMFAppearanceSettingsViewControllerDelegate?
+    
+    
+    static var disclosureText: String {
+        var text = WMFLocalizedString("default-theme-display-name", value: "Default", comment: "Default theme name presented to the user")
+        let currentAppTheme = UserDefaults.wmf_userDefaults().wmf_appTheme
+        
+        if currentAppTheme == Theme.sepia {
+            text = WMFLocalizedString("sepia-theme-display-name", value: "Sepia", comment: "Sepia theme name presented to the user")
+        } else if currentAppTheme == Theme.dark || currentAppTheme == Theme.darkDimmed {
+            text = WMFLocalizedString("dark-theme-display-name", value: "Dark", comment: "Dark theme name presented to the user")
+        }
+        
+        return text
+    }
+    
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.contentInset = UIEdgeInsets(top: 10, left: 0, bottom: 0, right: 0);
+        tableView.register(WMFSettingsTableViewCell.wmf_classNib(), forCellReuseIdentifier: WMFSettingsTableViewCell.identifier())
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorStyle = .none
+        sections = sectionsForAppearanceSettings()
+        apply(theme: self.theme)
+    }
+    
+    func defaultThemeDisplayName() -> String {
+        return WMFLocalizedString("default-theme-display-name", value: "Default", comment: "Default theme name presented to the user")
+    }
+    
+    func sepiaThemeDisplayName() -> String {
+        return WMFLocalizedString("sepia-theme-display-name", value: "Sepia", comment: "Sepia theme name presented to the user")
+    }
+    
+    func darkThemeDisplayName() -> String {
+        return WMFLocalizedString("dark-theme-display-name", value: "Dark", comment: "Dark theme name presented to the user")
+    }
```

Comment:
^ these should be removed, using the above suggestion, you could do `Theme.sepia.localizedName` etc.